### ISSUE TITLE
Flatpak can't access gstreamer plugin

### ISSUE THREAD

**Issue by Artim96** [hongquan/CoBang#76]
I just installed 0.10.1 from flathub, yet it can't access the camera due to these errors:
```
flatpak run vn.hoabinh.quan.CoBang

(gst-plugin-scanner:6): GStreamer-WARNING **: 18:13:33.395: Failed to load plugin '/app/lib/gstreamer-1.0/libgstva.so': /app/lib/libudev.so.1: version `LIBUDEV_247' not found (required by /usr/lib/x86_64-linux-gnu/libgudev-1.0.so.0)

(gst-plugin-scanner:6): GStreamer-WARNING **: 18:13:33.398: Failed to load plugin '/app/lib/gstreamer-1.0/libgstvideo4linux2.so': /app/lib/libudev.so.1: version `LIBUDEV_247' not found (required by /usr/lib/x86_64-linux-gnu/libgudev-1.0.so.0)

(__main__.py:3): cobang-CRITICAL **: 18:13:33.736: Failed to create Gst Pipeline. Error: gst_parse_error: no element "gtksink" (1)
```

Reading from an image works, but it doesn't show anything in "results", only in the raw results.

Running on Debian testing with Gnome 44 and on Wayland.

**Comment by Isolator70** [comment]
Same issue here on Linux Mint 21.2 Cinnamon Standard installation. It worked some time ago without problem, don't know when it stopped working.

> flatpak run vn.hoabinh.quan.CoBang
> Gtk-Message: 13:20:19.516: Failed to load module "xapp-gtk3-module"
> 
> (gst-plugin-scanner:4): GStreamer-WARNING **: 13:20:19.676: Failed to load plugin '/app/lib/gstreamer-1.0/libgstva.so': /app/lib/libudev.so.1: version `LIBUDEV_247' not found (required by /usr/lib/x86_64-linux-gnu/libgudev-1.0.so.0)
> 
> (gst-plugin-scanner:4): GStreamer-WARNING **: 13:20:19.678: Failed to load plugin '/app/lib/gstreamer-1.0/libgstvideo4linux2.so': /app/lib/libudev.so.1: version `LIBUDEV_247' not found (required by /usr/lib/x86_64-linux-gnu/libgudev-1.0.so.0)
> 
> (__main__.py:3): GStreamer-WARNING **: 13:20:20.011: Failed to load plugin '/app/lib/gstreamer-1.0/libgstopengl.so': /app/lib/libudev.so.1: version `LIBUDEV_247' not found (required by /usr/lib/x86_64-linux-gnu/libgudev-1.0.so.0)
> 
> (__main__.py:3): GStreamer-WARNING **: 13:20:20.016: Failed to load plugin '/app/lib/gstreamer-1.0/libgstgtk.so': /app/lib/libudev.so.1: version `LIBUDEV_247' not found (required by /usr/lib/x86_64-linux-gnu/libgudev-1.0.so.0)
> 
> (__main__.py:3): cobang-CRITICAL **: 13:20:20.021: Failed to create Gst Pipeline. Error: gst_parse_error: Kein Element »gtksink« (1)

**Comment by hongquan** [comment]
Thank you. I will try to reproduce it.

**Comment by hubertbanas** [comment]
CoBang version 0.10.1

Seeing the following:

```python
$ flatpak run vn.hoabinh.quan.CoBang

(gst-plugin-scanner:4): GStreamer-WARNING **: 14:48:56.002: Failed to load plugin '/app/lib/gstreamer-1.0/libgstvideo4linux2.so': /app/lib/libudev.so.1: version `LIBUDEV_247' not found (required by /usr/lib/x86_64-linux-gnu/libgudev-1.0.so.0)

(gst-plugin-scanner:4): GStreamer-WARNING **: 14:48:56.004: Failed to load plugin '/app/lib/gstreamer-1.0/libgstva.so': /app/lib/libudev.so.1: version `LIBUDEV_247' not found (required by /usr/lib/x86_64-linux-gnu/libgudev-1.0.so.0)
Gtk-Message: 14:48:56.255: Failed to load module "canberra-gtk-module"

(__main__.py:3): cobang-CRITICAL **: 14:48:56.264: Failed to create Gst Pipeline. Error: gst_parse_error: no element "gtksink" (1)
```

**Comment by hongquan** [comment]
Sorry for slow progress. Since v0.10.1, CoBang is broken with FlatHub. I'm trying to fix. It is difficult because it can only reproducible after releasing to FlatHub. It is Ok in development, though.

**Comment by hongquan** [comment]
@Artim96 , @Isolator70 , @hubertbanas It has been fixed in v0.10.3 and waiting to be listed in FlatHub. You can try the testing build as in https://github.com/flathub/vn.hoabinh.quan.CoBang/pull/9#issuecomment-1869690339

**Comment by Artim96** [comment]
> @Artim96 , @Isolator70 , @hubertbanas It has been fixed in v0.10.3 and waiting to be listed in FlatHub. You can try the testing build as in [flathub/vn.hoabinh.quan.CoBang#9 (comment)](https://github.com/flathub/vn.hoabinh.quan.CoBang/pull/9#issuecomment-1869690339)

Same error:
```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/app/lib/python3.11/site-packages/cobang/__main__.py", line 21, in <module>
    from .app import CoBangApplication
  File "/app/lib/python3.11/site-packages/cobang/app.py", line 33, in <module>
    from . import ui
  File "/app/lib/python3.11/site-packages/cobang/ui.py", line 16, in <module>
    from .resources import get_ui_filepath
  File "/app/lib/python3.11/site-packages/cobang/resources.py", line 8, in <module>
    import requests
  File "/app/lib/python3.11/site-packages/requests/__init__.py", line 43, in <module>
    import urllib3
  File "/app/lib/python3.11/site-packages/urllib3/__init__.py", line 13, in <module>
    from .connectionpool import HTTPConnectionPool, HTTPSConnectionPool, connection_from_url
  File "/app/lib/python3.11/site-packages/urllib3/connectionpool.py", line 40, in <module>
    from .response import HTTPResponse
  File "/app/lib/python3.11/site-packages/urllib3/response.py", line 162, in <module>
    class HTTPResponse(io.IOBase):
  File "/app/lib/python3.11/site-packages/urllib3/response.py", line 395, in HTTPResponse
    DECODER_ERROR_CLASSES += (brotli.error,)
                              ^^^^^^^^^^^^
AttributeError: module 'brotli' has no attribute 'error'
```

I uninstalled the regular version before installing it via that command you linked, though didn't remove user data with it.
